File: SingleNumber(136).js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    const cache = {}
    for(let i=0; i<nums.length; i++){
        if(cache[nums[i]] == undefined) cache[nums[i]] = 1;
        else cache[nums[i]]++;
    }
    
    return Object.entries(cache).filter(ele=>ele[1]==1).map(ele=>ele[0]);
};

File: SingleNumber(136).swift
class Solution {
    func singleNumber(_ nums: [Int]) -> Int {
        var dict: [Int: Int] = [:]

        for num in nums {
            if dict[num] == nil {
                dict[num] = 1
            } else {
                dict[num]! += 1
            }
        }

        return dict.filter { $0.value == 1 }.first!.key
    }
}
